### PR TITLE
Lightnet error and edge cases handling.

### LINKED ISSUE #512: [Lightnet] maxAttempts calculator based on the host hardware configuration and resources available for Docker Engine (if it is configurable)
https://github.com/o1-labs/zkapp-cli/pull/510#discussion_r1376603496

### LINKED ISSUE #544: [Lightnet] Better errors handling
Even though we have the `--debug` option to print the underlying commands execution stdout/stderr, it will be much better to explain what went wrong to the end used upon their first failure to not ask them repeat such commands again. Alongside with the error message we also need to print the underlying commands used.
For example, if `zk lightnet start` fails it will be good to see the exact `docker run` command in use as well as the actual failure reason (resources issues, ports already in use, etc.).

### PR DESCRIPTION
**Pull request by shimkiv**
Closes #512 
Closes #544 

---

image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,19 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## Unreleased
 
+### Breaking changes
+
+- Lightnet error and edge cases handling. [#597](https://github.com/o1-labs/zkapp-cli/pull/597)
+  - We removed common `--debug` CLI option in favor of [DEBUG](https://www.npmjs.com/package/debug) environment variable presence.
+  - From now on, in order to enable debug logging, you should set `DEBUG` environment variable to `<namespace>` value.
```
Comment (barriebyron) [inline CHANGELOG.md:24]:
```suggestion
  - From now on, in order to enable debug logging, you must set `DEBUG` environment variable to `<namespace>` value.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,19 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+    - Where `<namespace>` can be one of the following:
+      - `*`
+      - `zk:*`
+      - `zk:lightnet`
+    - Example: `DEBUG=zk:lightnet zk lightnet start`
+    - More details about the `DEBUG` environment variable can be found [here](https://www.npmjs.com/package/debug#usage).
+  - We are going to leverage the new debug logging capabilities with other parts of the CLI in the future.
```
Comment (barriebyron) [inline CHANGELOG.md:31]:
```suggestion
    - More details about the `DEBUG` environment variable can be found [here](https://www.npmjs.com/package/debug#usage).
  - We are going to leverage the new debug logging capabilities with other parts of the CLI in the future.
```
```suggestion
    - For details about the `DEBUG` environment variable, see Debug [Usage](https://www.npmjs.com/package/debug#usage).
  - This improved debug logging capabilities can be leveraged with other parts of the zkApp CLI in the future.
```

**Review by ymekuria — APPROVED** [review]
Nice! I just had a minor comment/suggestion.

File: src/lib/helpers.js [hunk 3]
```diff
@@ -43,4 +44,45 @@ export async function isMinaGraphQlEndpointAvailable(endpoint) {
...
+  const checks = ports.map((port) => checkLocalPortAvailability(port));
+  const results = await Promise.all(checks);
+  const busyPorts = results
+    .filter((result) => result.busy)
+    .map((result) => result.port);
+  if (busyPorts.length > 0) {
+    return {
+      error: true,
+      message:
+        `The following local ports are required but unavailable at this time: ${busyPorts.join(', ')}`.trim(),
```
[the remark this answers is not in the document]
Comment (shimkiv) [reply src/lib/helpers.js:81]: Thank you @ymekuria it definitely will improve dx. I'll update message.

**Pull request merged**